Datagrid/ListMapper.php

```diff
@@ -124,6 +124,34 @@ public function add($name, $type = null, array $fieldDescriptionOptions = array(
             );
         }
 
+        // NEXT_MAJOR: Remove first "if" statement (when requirement of Symfony is >= 2.8)
+        if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) { // symfony >= 2.8
+            $choices = $fieldDescription->getOption('choices');
+            if (($type == 'choice' || $type == 'Symfony\Component\Form\Extension\Core\Type\ChoiceType') && $choices) {
```

Comment:
`$type` variable should not be checked against `ChoiceType`, as fieldTypes are strings as you can see at https://sonata-project.org/bundles/admin/3-x/doc/reference/field_types.html.